mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Text.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\EmailEditor\Integrations\Core\Renderer\Blocks;
+
+use MailPoet\EmailEditor\Engine\SettingsController;
+
+/**
+ * This renderer covers both core/paragraph and core/heading blocks
+ */
+class Text extends AbstractBlockRenderer {
+  protected function renderContent(string $blockContent, array $parsedBlock, SettingsController $settingsController): string {
+    $blockContent = $this->adjustStyleAttribute($blockContent);
+    return str_replace('{heading_content}', $blockContent, $this->getBlockWrapper($blockContent, $parsedBlock));
+  }
+
+  /**
+   * Based on MJML <mj-text>
+   */
+  private function getBlockWrapper($blockContent, array $parsedBlock): string {
+    $html = new \WP_HTML_Tag_Processor($blockContent);
+    $classes = '';
+    if ($html->next_tag()) {
+      $classes = $html->get_attribute('class') ?? '';
+    }
+
+    $blockStyles = $this->getStylesFromBlock([
+      'color' => $parsedBlock['attrs']['style']['color'] ?? [],
+      'spacing' => $parsedBlock['attrs']['style']['spacing'] ?? [],
+      'typography' => $parsedBlock['attrs']['style']['typography'] ?? [],
+    ]);
+
+    $styles = [
+      'min-width' => '100%', // prevent Gmail App from shrinking the table on mobile devices
+    ];
+
+    $styles['text-align'] = 'left';
+    if (isset($parsedBlock['attrs']['textAlign'])) {
+      $styles['text-align'] = $parsedBlock['attrs']['textAlign'];
+    } elseif (in_array($parsedBlock['attrs']['align'] ?? null, ['left', 'center', 'right'])) {
+      $styles['text-align'] = $parsedBlock['attrs']['align'];
+    }
+
+    $compiledStyles = $this->compileCss($blockStyles['declarations'], $styles);
+
+    return '
+      <!--[if mso | IE]><table align="left" role="presentation" border="0" cellpadding="0" cellspacing="0" width="100%"><tr><td><![endif]-->
+          <table
+            role="presentation"
+            border="0"
+            cellpadding="0"
+            cellspacing="0"
+            style="min-width: 100%;"
+            width="100%"
+          >
+            <tr>
+              <td class="' . esc_attr($classes) . '" style="' . esc_attr($compiledStyles) . '" align="' . esc_attr($styles['text-align'] ?? 'left') . '">
+                {heading_content}
+              </td>
+            </tr>
+          </table>
+
+      <!--[if mso | IE]></td></tr></table><![endif]-->
+    ';
+  }
+
+  /**
+   * 1) We need to remove padding because we render padding on wrapping table cell
+   * 2) We also need to replace font-size to avoid clamp() because clamp() is not supported in many email clients.
+   * The font size values is automatically converted to clamp() when WP site theme is configured to use fluid layouts.
+   * Currently (WP 6.5), there is no way to disable this behavior.
+   */
```

Comment:
@mikejolley Actually, I don't know how to do that without manipulating the original HTML. If user sets custom values for padding and font size the input HTML we get to the renderer looks like this:
```html
<h4 class="wp-block-heading alignfull has-background" style="background-color:#8cb3cb;padding-top:29px;padding-right:19px;padding-bottom:29px;padding-left:19px;font-size:clamp(21.027px, 1.314rem + ((1vw - 3.2px) * 1.351), 34px);">Heading with custom values</h4>
```
The thing is, we need to apply padding and background CSS on the table wrapper to get it rendered in Outlook, but we can't keep it on the original HTML because clients that support styles for paragraphs and headings would render the padding on both.

Maybe I just can't see a better approach. Any idea?

Reply:
> I would have suggested re-rendering the element, but since this code is shared across headings and paragraphs the current solution is better.

@mikejolley I was trying that approach. I could get the tag using the WP_HTML_Tag_Processor and then the inner HTML, but this doesn't seem to be supported by the WP HTML API, so I would probably have to use PHP Dom so it didn't look like a simpler/cleaner solution than what we have in place.

> You can ignore my comment.

In that case. Could you please take another look at the PR? I added two commits.